From 25cdebeaa3e57aef9c216823ab39dae5e5f634ee Mon Sep 17 00:00:00 2001
From: Dong Chuanxiao <chuanxiao.dong@intel.com>
Date: Fri, 19 Jul 2013 10:06:36 +0800
Subject: [PATCH 239/429] mmc: host: use bit4 of power control to do HW reset
 for CLV

CLV cannot use normal GPIO to do hardware reset. And the GPIO is
configured to be alternate function 1, and bit4 of the power control
can be used to do the HW reset. This is specific to CLV eMMC host
controller

Signed-off-by: Ning Feiyi <feiyix.ning@intel.com>
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/host/sdhci-pci.c | 1 +
 drivers/mmc/host/sdhci.c     | 2 ++
 drivers/mmc/host/sdhci.h     | 1 +
 3 files changed, 4 insertions(+)

diff --git a/drivers/mmc/host/sdhci-pci.c b/drivers/mmc/host/sdhci-pci.c
index fa266360dbe4..2af6d660f501 100644
--- a/drivers/mmc/host/sdhci-pci.c
+++ b/drivers/mmc/host/sdhci-pci.c
@@ -1538,6 +1538,7 @@ static void sdhci_pci_hw_reset(struct sdhci_host *host)
 {
 	struct sdhci_pci_slot *slot = sdhci_priv(host);
 	int rst_n_gpio = slot->rst_n_gpio;
+	u8 pwr;
 
 	if (gpio_is_valid(rst_n_gpio)) {
 		gpio_set_value_cansleep(rst_n_gpio, 0);
diff --git a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
index eab81d62082b..ba533ac2dee5 100644
--- a/drivers/mmc/host/sdhci.c
+++ b/drivers/mmc/host/sdhci.c
@@ -1884,9 +1884,11 @@ static void sdhci_hw_reset(struct mmc_host *mmc)
 	struct sdhci_host *host = mmc_priv(mmc);
 
 	if (host->ops && host->ops->hw_reset) {
+		sdhci_runtime_pm_get(host);
 		sdhci_acquire_ownership(mmc);
 		host->ops->hw_reset(host);
 		sdhci_release_ownership(mmc);
+		sdhci_runtime_pm_put(host);
 	}
 }
 
diff --git a/drivers/mmc/host/sdhci.h b/drivers/mmc/host/sdhci.h
index a8ae0bdd1e27..526babe60591 100644
--- a/drivers/mmc/host/sdhci.h
+++ b/drivers/mmc/host/sdhci.h
@@ -89,6 +89,7 @@
 #define  SDHCI_POWER_180	0x0A
 #define  SDHCI_POWER_300	0x0C
 #define  SDHCI_POWER_330	0x0E
+#define	 SDHCI_HW_RESET		0x10
 
 #define SDHCI_BLOCK_GAP_CONTROL	0x2A
 
-- 
2.37.3

